api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java

```diff
@@ -164,11 +164,26 @@ public String getTagsAsString() {
      * @return a a list of Tag objects
      */
     public List<Tag> getTagsAsList() {
-        List<Tag> list = new ArrayList<Tag>();
+        List<Tag> list = new ArrayList<>();
         this.tags.forEach((key, value) -> list.add(new Tag(key, value)));
         return Collections.unmodifiableList(list);
     }
 
+    /**
+     * Gets the list of tags as an array of {@link Tag} objects.
+     *
+     * @return An array of tags
+     */
+    public Tag[] getTagsAsArray() {
```

Comment:
Could one use `tags.toArray(new Tag[tags.size()]` ?

Reply:
unfortunately the `tags` here is a `Map<String, String>`, not a `List<Tag>` so this is not possible without first converting it to a list, which would be more expensive than this

Reply:
I feel like getting the tags from one metricId to use to form a new metric is a bit of an edge case -- so I don't think I'd want to give up the memory for that.